src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -12,13 +23,374 @@
+                    lambda d: d[d <= d.quantile(outlier_percentile)].mean(),
+                    lambda d: d[d <= d.quantile(outlier_percentile)].std(),
+                ]
+            )
+            .transpose()
+        )
+        overall_stats.columns = ["mean", "std", "mean_trimmed", "std_trimmed"]
+        overall_stats["trim_change_mean"] = (
+            overall_stats["mean_trimmed"] - overall_stats["mean"]
+        ).abs() / overall_stats["mean"]
+        overall_stats["trim_change_std"] = (
+            overall_stats["std_trimmed"] - overall_stats["std"]
+        ).abs() / overall_stats["std"]
+        self._raw_data_stats = overall_stats
+
+    def get_dataframe(
+        self,
+        input_data: pd.DataFrame = None,
+        show_population_pct: bool = True,
+        simulated_values: list = None,
+        append_stats: bool = False,
+    ) -> pd.DataFrame:
+        """
+        obtain data for sample size curves
+
+        Args:
+            input_data (pd.DataFrame, optional): Input data used to generate results,
+                used to generate statistics. Defaults to None.
+            show_population_pct (bool, optional): If true, percentage of population
+                will be used in output rather than raw counts. Defaults to True.
+            simulated_values (list, optional): List of simulated values to subset
+                output to. If None, all values will be included. Defaults to None.
+            append_stats (bool, optional): _description_. If True, summary statistics on
+                the raw metrics are provided. Defaults to False.
+            highlight_lessthan (list[tuple], optional): List of
+                (<cutoff (float)>, <color (string)>) tuples.
+                Sample size entries less than <cutoff> will be highlighted in <color>
+                in the displayed table. Defaults to None.
+
+        Raises:
+            ValueError: Raises error if append_stats is True but no input_data is
+                        provided
+
+        Returns:
+            pd.DataFrame : dataframe consolidating data
+        """
+        # choose which column to output
+        if show_population_pct:
+            subset_col = "population_percent_per_branch"
+        else:
+            subset_col = "sample_size_per_branch"
+
+        # reformate raw_df to make different simulatd values into columns
+        if simulated_values is None:
+            simulated_values = self._params["simulated_values"]
+            pretty_df = self.raw_df[subset_col].unstack()
+        else:
+            pretty_df = self.raw_df[subset_col].unstack()[simulated_values]
+
+        # get stats if input_data is non-null and append_stats is True
+        if append_stats:
+            if input_data is None:
+                raise ValueError("append_stats is true but no raw data was provided")
+            self.set_raw_data_stats(input_data)
+            pretty_df = pd.concat([pretty_df, self._raw_data_stats], axis="columns")
```

Comment:
Similarly, in keeping with `get_dataframe` for the other result types, maybe appending stats should happen in `get_styled_dataframe`, but the output of this method just contains the main results returned by the sizing tool